### DISCUSSION TITLE
Should users be forced to write out the full fnctl argument in `regress`?

### DISCUSSION THREAD

**Discussion by taylorokonek** [statdivlab/rigr#53] [Ideas]
Currently, users can enter a shortened character string for the fnctl argument in `regress`, and so long as it corresponds to a unique substring, the function will run. As an example, a user could simply set `fnctl = "g"`, and the function would take this to mean `fnctl = "geometric mean"`. Personally, I don't think there's a huge harm in forcing users to spell out the fnctl argument. Does anyone have any insight as to why this particular functionality was introduced (@bdwilliamson)?

**Comment by bdwilliamson** [comment]
I don’t remember why we did that, and also think it’s fine to force people to write out the entire functional.
